contrib/epee/include/net/net_utils_base.h

```diff
@@ -94,17 +94,51 @@ namespace net_utils
 		END_KV_SERIALIZE_MAP()
 	};
 
-	inline bool operator==(const ipv4_network_address& lhs, const ipv4_network_address& rhs) noexcept
+	class ipv4_network_subnet
+	{
+		uint32_t m_ip;
+		uint8_t m_mask;
+
+	public:
+		constexpr ipv4_network_subnet() noexcept
+			: ipv4_network_subnet(0, 0)
+		{}
+
+		constexpr ipv4_network_subnet(uint32_t ip, uint8_t mask) noexcept
+			: m_ip(ip), m_mask(mask) {}
+
+		bool equal(const ipv4_network_subnet& other) const noexcept;
+		bool less(const ipv4_network_subnet& other) const noexcept;
+		constexpr bool is_same_host(const ipv4_network_subnet& other) const noexcept
+		{ return ip() == other.ip(); }
```

Comment:
Is it easier to just compare the ip and masks without the extra bit twiddling?

Reply:
I suppose it could if you were to make the underlying ip canonical on ctor. It's just the same thing really, just in different places.